From 14632e11da35720e2b32d155ee01c650065fbb1e Mon Sep 17 00:00:00 2001
From: Chun-wei Fan <fanchunwei@src.gnome.org>
Date: Fri, 16 Apr 2021 11:22:27 +0800
Subject: [PATCH 1/3] GDK: Fix builds against latest stable GLib with MSVC

GLib-2.68.x now considers warning C4098 ('void' function returning a value) as
an error, so avoid doing that.
---
 gdk/gdkclipboard.c       | 20 ++++++++++----------
 gdk/gdkcontentprovider.c |  4 ++--
 gdk/gdktexture.c         |  2 +-
 gdk/gdktoplevel.c        |  2 +-
 4 files changed, 14 insertions(+), 14 deletions(-)

diff --git a/gdk/gdkclipboard.c b/gdk/gdkclipboard.c
index df20366d53..c3a4759c79 100644
--- a/gdk/gdkclipboard.c
+++ b/gdk/gdkclipboard.c
@@ -554,19 +554,19 @@ gdk_clipboard_store_async (GdkClipboard        *clipboard,
 
   if (priv->local)
     {
-      return GDK_CLIPBOARD_GET_CLASS (clipboard)->store_async (clipboard,
-                                                               io_priority,
-                                                               cancellable,
-                                                               callback,
-                                                               user_data);
+      GDK_CLIPBOARD_GET_CLASS (clipboard)->store_async (clipboard,
+                                                        io_priority,
+                                                        cancellable,
+                                                        callback,
+                                                        user_data);
     }
   else
     {
-      return gdk_clipboard_store_default_async (clipboard,
-                                                io_priority,
-                                                cancellable,
-                                                callback,
-                                                user_data);
+      gdk_clipboard_store_default_async (clipboard,
+                                         io_priority,
+                                         cancellable,
+                                         callback,
+                                         user_data);
     }
 }
 
diff --git a/gdk/gdkcontentprovider.c b/gdk/gdkcontentprovider.c
index b01cb43e93..15e183a178 100644
--- a/gdk/gdkcontentprovider.c
+++ b/gdk/gdkcontentprovider.c
@@ -377,7 +377,7 @@ gdk_content_provider_attach_clipboard (GdkContentProvider *provider,
   g_return_if_fail (GDK_IS_CONTENT_PROVIDER (provider));
   g_return_if_fail (GDK_IS_CLIPBOARD (clipboard));
 
-  return GDK_CONTENT_PROVIDER_GET_CLASS (provider)->attach_clipboard (provider, clipboard);
+  GDK_CONTENT_PROVIDER_GET_CLASS (provider)->attach_clipboard (provider, clipboard);
 }
 
 void
@@ -387,5 +387,5 @@ gdk_content_provider_detach_clipboard (GdkContentProvider *provider,
   g_return_if_fail (GDK_IS_CONTENT_PROVIDER (provider));
   g_return_if_fail (GDK_IS_CLIPBOARD (clipboard));
 
-  return GDK_CONTENT_PROVIDER_GET_CLASS (provider)->detach_clipboard (provider, clipboard);
+  GDK_CONTENT_PROVIDER_GET_CLASS (provider)->detach_clipboard (provider, clipboard);
 }
diff --git a/gdk/gdktexture.c b/gdk/gdktexture.c
index 07235eb97c..3a22d2133e 100644
--- a/gdk/gdktexture.c
+++ b/gdk/gdktexture.c
@@ -446,7 +446,7 @@ gdk_texture_download_area (GdkTexture         *texture,
   g_assert (area->x + area->width <= texture->width);
   g_assert (area->y + area->height <= texture->height);
 
-  return GDK_TEXTURE_GET_CLASS (texture)->download (texture, area, data, stride);
+  GDK_TEXTURE_GET_CLASS (texture)->download (texture, area, data, stride);
 }
 
 /**
diff --git a/gdk/gdktoplevel.c b/gdk/gdktoplevel.c
index 1cf8188b8b..ac89e61671 100644
--- a/gdk/gdktoplevel.c
+++ b/gdk/gdktoplevel.c
@@ -364,7 +364,7 @@ gdk_toplevel_focus (GdkToplevel *toplevel,
 {
   g_return_if_fail (GDK_IS_TOPLEVEL (toplevel));
 
-  return GDK_TOPLEVEL_GET_IFACE (toplevel)->focus (toplevel, timestamp);
+  GDK_TOPLEVEL_GET_IFACE (toplevel)->focus (toplevel, timestamp);
 }
 
 /**
-- 
GitLab


From 8df32e93bea21a67d41ded974f991e9c67e942b6 Mon Sep 17 00:00:00 2001
From: Chun-wei Fan <fanchunwei@src.gnome.org>
Date: Fri, 16 Apr 2021 11:33:00 +0800
Subject: [PATCH 2/3] GSK: Fix builds against latest stable GLib with MSVC

GLib-2.68.x now considers warning C4098 ('void' function returning a value) as
an error, so avoid doing that.
---
 gsk/gskrendernode.c | 4 +++++++-------
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/gsk/gskrendernode.c b/gsk/gskrendernode.c
index fb2ddd3962..ed1a1eb7f8 100644
--- a/gsk/gskrendernode.c
+++ b/gsk/gskrendernode.c
@@ -547,4 +547,4 @@ gsk_render_node_diff (GskRenderNode  *node1,
   if (_gsk_render_node_get_node_type (node1) != _gsk_render_node_get_node_type (node2))
-    return gsk_render_node_diff_impossible (node1, node2, region);
+    gsk_render_node_diff_impossible (node1, node2, region);

-  return GSK_RENDER_NODE_GET_CLASS (node1)->diff (node1, node2, region);
+  GSK_RENDER_NODE_GET_CLASS (node1)->diff (node1, node2, region);
}
 
 /**
-- 
GitLab


From 153019a794cec44e66024dff55ccfebb66ee94fb Mon Sep 17 00:00:00 2001
From: Chun-wei Fan <fanchunwei@src.gnome.org>
Date: Fri, 16 Apr 2021 11:36:18 +0800
Subject: [PATCH 3/3] GTK: Fix builds against latest stable GLib with MSVC

GLib-2.68.x now considers warning C4098 ('void' function returning a value) as
an error, so avoid doing that.
---
 gtk/gtkcssimagepaintable.c | 6 +++---
 gtk/gtknative.c            | 4 ++--
 gtk/gtkshortcutaction.c    | 2 +-
 3 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/gtk/gtkcssimagepaintable.c b/gtk/gtkcssimagepaintable.c
index d4056ab263..d45955c5df 100644
--- a/gtk/gtkcssimagepaintable.c
+++ b/gtk/gtkcssimagepaintable.c
@@ -66,9 +66,9 @@ gtk_css_image_paintable_snapshot (GtkCssImage *image,
 {
   GtkCssImagePaintable *paintable = GTK_CSS_IMAGE_PAINTABLE (image);
 
-  return gdk_paintable_snapshot (get_paintable (paintable),
-                                 snapshot,
-                                 width, height);
+  gdk_paintable_snapshot (get_paintable (paintable),
+                          snapshot,
+                          width, height);
 }
 
 static GtkCssImage *
diff --git a/gtk/gtknative.c b/gtk/gtknative.c
index 546b088a43..7856d4f486 100644
--- a/gtk/gtknative.c
+++ b/gtk/gtknative.c
@@ -102,7 +102,7 @@ gtk_native_layout (GtkNative *self,
                    int        width,
                    int        height)
 {
-  return GTK_NATIVE_GET_IFACE (self)->layout (self, width, height);
+  GTK_NATIVE_GET_IFACE (self)->layout (self, width, height);
 }
 
 static void
@@ -258,7 +258,7 @@ gtk_native_get_surface_transform (GtkNative *self,
   g_return_if_fail (x != NULL);
   g_return_if_fail (y != NULL);
 
-  return GTK_NATIVE_GET_IFACE (self)->get_surface_transform (self, x, y);
+  GTK_NATIVE_GET_IFACE (self)->get_surface_transform (self, x, y);
 }
 
 /**
diff --git a/gtk/gtkshortcutaction.c b/gtk/gtkshortcutaction.c
index 7870b0465c..62d213911d 100644
--- a/gtk/gtkshortcutaction.c
+++ b/gtk/gtkshortcutaction.c
@@ -132,7 +132,7 @@ gtk_shortcut_action_print (GtkShortcutAction *self,
   g_return_if_fail (GTK_IS_SHORTCUT_ACTION (self));
   g_return_if_fail (string != NULL);
 
-  return GTK_SHORTCUT_ACTION_GET_CLASS (self)->print (self, string);
+  GTK_SHORTCUT_ACTION_GET_CLASS (self)->print (self, string);
 }
 
 /**
-- 
GitLab

